packages/core-compat-api/src/components/SystemIcon.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ComponentProps } from 'react';
+import { useApp, IconComponent } from '@backstage/core-plugin-api';
+import { compatWrapper } from '../compatWrapper';
+
+/**
+ * @public
+ * Props for the SystemIcon component.
+ */
+export type SystemIconProps = ComponentProps<IconComponent> & {
+  // The id of the system icon to render, if provided as an array, the first icon found will be rendered.
+  keys: string | string[];
```

Comment:
Ah in the end I don't think we really wanna go with this one either. I think there's a pretty big risk that being able to list multiple keys ends up being really confusing to adopters that are configuring icons. It's kinda bad enough that adding icons for certain keys can replace icons in other plugins, but if you also need to juggle the case where there may be multiple different keys that affect certain plugins and try to find the correct key to use for a certain update, it all gets really messy.

tl;dr: thinking just `key: string` for now